wagtail/blocks/stream_block.py

```diff
@@ -304,6 +309,51 @@ def get_prep_value(self, value):
             # round-trips to the full data representation and back)
             return value.get_prep_value()
 
+    def normalize(self, value):
+        if value is None or value == "":
+            return self.empty_value()
+        elif isinstance(value, StreamValue):
+            return value
+        elif isinstance(value, str):
+            try:
+                unpacked_value = json.loads(value)
+            except ValueError:
+                # value is not valid JSON; most likely, this field was previously a
+                # rich text field before being migrated to StreamField, and the data
+                # was left intact in the migration. Return an empty stream instead
+                # (but keep the raw text available as an attribute, so that it can be
+                # used to migrate that data to StreamField)
+                return self.empty_value(self, [], raw_text=value)
```

Comment:
Worth noting that now StreamField always uses JSONField as the underlying database field, there is no circumstance where StreamField will be passed a string - JSON or otherwise - other than user code doing that explicitly with `page.body = "..."`. Indeed, the stated use-case for `raw_text` is no longer possible because converting a text field (containing non-JSON text) to StreamField will fail with a check constraint error. I think it would make sense to deprecate this.

Reply:
Scratch that - it does happen when unpacking revision records (at least, ones created prior to the migration to JSONField).